scheduler/duration_based_host_allocator.go

```diff
@@ -264,7 +263,7 @@ func computeDurationBasedNumNewHosts(scheduledTasksDuration,
 		numExistingDistroHosts
 
 	// duration based number of new hosts needed
-	numNewHosts = int(math.Ceil(durationBasedNumNewHostsNeeded))
+	numNewHosts = int(math.Floor(durationBasedNumNewHostsNeeded))
```

Comment:
This could mean that if calculate we need .8 hosts, we end up spawning 0. If `0 < durationBasedNumNewHostsNeeded < 1`, we should return 1.

Reply:
So maybe I think this is the right thing. (to be fair, this slipped in.) 

The reason why this is a float at all is because if a task appears in more than one queue, we multiply it's time requirements by .8 for the non-primary (whatever that means) distros that the task appears in. (There's a constant at the top of the file) 

I suspect that this is a source of over-provisioning of hosts: the only thing that we have to worry about (I think) is cases where we might have provisioned one host but didn't provision any. I added a catch for this.